### ISSUE TITLE
Can git-filter-repo rescue me from bad svn decisions?

### ISSUE THREAD

**Issue by jerkstorecaller** [newren/git-filter-repo#190]
I'm doing an svn to git migration. svn has a lot of flexibility in its externals (shortcuts/symlinks to another svn directory or file, pegged at a specific or HEAD revision), way beyond what git can support with submodules. Therefore, no git importer can successfully deal with externals, they just omit them completely during the migration.

I'm dealing with a project which had this:
```
/trunk
	/sharedcode/
		/sharedfunctionality1/
			...
/branches/somebranch/
	/app/
		/src/
			/dep1/ -> points at /trunk/sharedcode/sharedfunctionality1/
/branches/anotherbranch/
	/app/
		/src/
			/dep1/ -> points at /trunk/sharedcode/sharedfunctionality1/
```

After doing the migration to git, I have 3 git branches (actually more, but this will suffice): trunk, somebranch, and anotherbranch. Due to what I explained above, somebranch and anotherbranch are altogether missing /dep1, so their structure is now:
```
/app/
	/src/
```

And of course, since dep1/ is missing, builds are broken.

Could I use filter-repo to rewrite the history, so that dependency1/ is somehow integrated back into each branch? Note that I'm not just interested in having the tip of the branch build, or I would just do a straight up directory copy, commit it, and call it a day. I'd like history to remain valid like I was using svn. So if I checkout a historical somebranch commit from last year, it would have the dep1/ code from that same day last year.

I won't be keeping the trunk branch once this migration is done. So I don't need to maintain any links between branches and trunk, just move the history correctly.

TLDR: I a branch containing a dependency that another branch needs, due to svn migration. It shouldn't have been an external dependency. So I would like a branch's /path/to/dependency to be inserted into anotherbranch at a /specific/path/, even for older commits (basically as soon as dependency was introduced).
